### PR TITLE
NXP: Add support for Kinetis KE1xF SoC series and TWR-KE18F development board

### PR DESCRIPTION
**Pull request by henrikbrixandersen**
This patch series adds support for the NXP Kinetis KE1xF SoC series and the TWR-KE18F development board through the following commits:

* boards: twr-ke18f: add support for the NXP TWR-KE18F board
* soc: arm: ke1xf: add NXP Kinetis KE1xF SoC series support
* drivers: clock_control_mcux_sim: add support for clkout
* drivers: serial: uart_mcux_lpuart: add support for LPUARTs with two IRQs
* clock_control: introduce mcux pcc driver
* ext: hal: nxp: mcux: add device files for KE1xF series SoCs

### CODE REVIEW

**Comment by henrikbrixandersen** [comment]
PR adding support for further peripherals (SPI, I2C, WDOG, CAN, ADC12, ...) will follow once ready.

File: boards/arm/twr_ke18f/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,211 @@
+.. _twr_ke18f:
+
+NXP TWR-KE18F
+#############
+
+Overview
+********
+
+The TWR-KE18F is a development board for platform for the NXP Kinetis
```
Comment (dbkinder) [inline boards/arm/twr_ke18f/doc/index.rst]: How about:
```
The TWR-KE18F is a development board for NXP Kinetis
KE1xF 32-bit MCU-based platforms.  The onboard OpenSDAv2 serial and
debug adapter, running an open source bootloader, offers options for
serial communication, flash programming, and run-control debugging.
```

File: boards/arm/twr_ke18f/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+- RGB LED
+- Four user LEDs
+- Two user push-buttons
+- Potentiometer
+- Thermistor
+- Infrared port (IrDA)
+- CAN pin header
+- Flex I/O pin header
+
+For more information about the KE1xF SoC and the TWR-KE18F board:
```
Comment (dbkinder) [inline boards/arm/twr_ke18f/doc/index.rst]:
```suggestion
For more information about the KE1xF SoC and the TWR-KE18F board,
see these NXP reference documents:
```

File: boards/arm/twr_ke18f/doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
++-----------+------------+-------------------------------------+
+| GPIO      | on-chip    | gpio                                |
++-----------+------------+-------------------------------------+
+| UART      | on-chip    | serial port-polling;                |
+|           |            | serial port-interrupt               |
++-----------+------------+-------------------------------------+
+| FLASH     | on-chip    | soc flash                           |
++-----------+------------+-------------------------------------+
+
+The default configuration can be found in the defconfig file:
```
Comment (dbkinder) [inline boards/arm/twr_ke18f/doc/index.rst]: This renders better as:
```
The default configuration can be found in the defconfig file:
``boards/arm/twr_ke18f/twr_ke18f_defconfig``.
```

**Review by dbkinder — CHANGES_REQUESTED** [review]
Doc changes look good, with a few suggested edits

File: boards/arm/twr_ke18f/doc/index.rst (hunk 1 above)
Comment (henrikbrixandersen) [reply boards/arm/twr_ke18f/doc/index.rst]: @dbkinder I appreciate your review. I have made the suggested changes.

File: ext/hal/nxp/mcux/README [hunk 4]
```diff
@@ -24,6 +24,9 @@ Status:
    MIMXRT1021	SDK 2.5.0 (2018-12-17)		REL_2.5.0_REL9_RFP_RC3_7_1
    MIMXRT1064	SDK 2.4.0 (2018-10-22)		REL_SDK_2.4.0_RT1064_RFP_INTERNAL
    MK64F12	SDK 2.5.0 (2018-12-17)		REL_2.5.0_REL9_RFP_RC3_7_1
+   MKE14F16	SDK 2.5.0 (2018-12-17)		REL_2.5.0_REL9_RFP_RC3_7_1
+   MKE16F16	SDK 2.5.0 (2018-12-17)		REL_2.5.0_REL9_RFP_RC3_7_1
+   MKE18F16	SDK 2.5.0 (2018-12-17)		REL_2.5.0_REL9_RFP_RC3_7_1
```
Comment (henrikbrixandersen) [inline ext/hal/nxp/mcux/README]: @MaureenHelm I am not sure if the tag is correct for these imports. Can I see the tag from the generated MCUX SDK archive somehow?

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Wow, great work! Just a couple of nits, and please confirm the SDK version.

File: drivers/clock_control/clock_control_mcux_pcc.c [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+}
+
+static const struct clock_control_driver_api mcux_pcc_api = {
+	.on = mcux_pcc_on,
+	.off = mcux_pcc_off,
+	.get_rate = mcux_pcc_get_rate,
+};
+
+#if defined(DT_MCUX_PCC_0_NAME)
+static struct mcux_pcc_config mcux_pcc0_config = {
```
[the remark this answers is not in the document]
Comment (henrikbrixandersen) [reply drivers/clock_control/clock_control_mcux_pcc.c]: Thank you for the swift review. I have made this const, good catch.

File: dts/arm/nxp/nxp_ke1xf.dtsi [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+			reg = <0x4006b000 0x1000>;
+			interrupts = <33 0>, <34 0>;
+			interrupt-names = "transmit", "receive";
+			clocks = <&pcc 0x1ac>;
+			label = "UART_1";
+			status = "disabled";
+		};
+
+		uart2: uart@4006c000 {
+			compatible = "nxp,kinetis-uart";
```
[the remark this answers is not in the document]
Comment (henrikbrixandersen) [reply dts/arm/nxp/nxp_ke1xf.dtsi]: Fixed, thanks.

File: ext/hal/nxp/mcux/README (hunk 4 above)
Comment (henrikbrixandersen) [reply ext/hal/nxp/mcux/README]: Screenshot 2019-05-10 at 19 37 26

@MaureenHelm My list of recent SDKs does not mention the tag. Perhaps I should leave them blank (or `-` in the `README` file?

Comment (MaureenHelm) [reply ext/hal/nxp/mcux/README]: Let me check with the SDK team to find out why the tag isn't there

**Review by dbkinder — APPROVED** [review]
+1 for docs, thanks!

File: ext/hal/nxp/mcux/README (hunk 4 above)
Comment (MaureenHelm) [reply ext/hal/nxp/mcux/README]: It turns out that only employee accounts can see the tag. Instead, please use the "revision" attribute from `TWR-KE18F_manifest_v3_4.xml`:
```xml
<ksdk id="MCUXpresso250" name="MCUXpresso250" version="2.5.0" revision="150 2018-12-17"/>
```

Comment (henrikbrixandersen) [reply ext/hal/nxp/mcux/README]: @MaureenHelm Thank you for following up on this. I have changed the `README` file to list the `revision` attribute values.

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**